### PR TITLE
fix: liq 716 ledger to regular swap

### PR DESCRIPTION
**Pull request by bradleySuira**
# :books: Linear Ticket :books:

- [LIQ-796](https://linear.app/liquality/issue/LIQ-796/ledger-connection-show-available-balances-by-derivation-path)
- [LIQ-714](https://linear.app/liquality/issue/LIQ-714/ledger-to-ledger-swap-is-not-working)

### CODE REVIEW

File: src/utils/ledger-bridge-provider/BitcoinLedgerBridgeApp.js [hunk 1]
```diff
@@ -1,46 +1,45 @@
...
+    return this.callToBridge({
       action: 'createPaymentTransactionNew',
       execMode: ExecutionMode.Async,
       payload
     })
   }
 
   async serializeTransactionOutputs (...payload) {
-    return await this.callToBridge({
+    const result = await this.callToBridge({
```
Comment (harshjv) [inline src/utils/ledger-bridge-provider/BitcoinLedgerBridgeApp.js:38]: for consistency, can we also do `return this.callToBridge({` here?

**Review by harshjv — CHANGES_REQUESTED** [review]

File: src/store/actions/hardware-wallet/getLedgerAccounts.js [hunk 2]
```diff
@@ -1,17 +1,49 @@
...
     return account.chain === chain
   })
 
   const pageIndexes = [...Array(numAccounts || 5).keys()].map(i => i + startingIndex)
+  const xpubVersion = getXPubVersion(network)
+  let xPub = null
+
+  // For BTC we need to get the xPub to store it
+  // and then we don't need to call again the ledger device to get all addresses
+  if (chain === ChainId.Bitcoin) {
```
[the remark this answers is not in the document]
Comment (bradleySuira) [reply src/store/actions/hardware-wallet/getLedgerAccounts.js:28]: agreed, can be something like `isUtxoChain()` a version of `isEthreumChain`

File: src/utils/ledger-bridge-provider/BitcoinLedgerBridgeApp.js (hunk 1 above)
Comment (bradleySuira) [reply src/utils/ledger-bridge-provider/BitcoinLedgerBridgeApp.js:38]: fixed, thanks!

File: src/store/getters.js [hunk 3]
```diff
@@ -57,15 +57,28 @@ export default {
...
       const _accountType = account?.type || accountType
       const _accountIndex = account?.index || accountIndex
-      const { chain } = getters.cryptoassets[asset]
+      const { chain } = cryptoassets[asset]
       let derivationPath
+      let xPub = null
       // when we ask for ledger accounts from the ledger device we don't have the derivation path
       // the !account doesn't exist in this case or if we call the getter with accountId equals to null
+      if (asset === 'BTC') {
+        console.log(
```
[the remark this answers is not in the document]
Comment (bradleySuira) [reply src/store/getters.js]: fixed

**Review by harshjv — APPROVED** [review]
LGTM

**Pull request merged**